File: app/src/main/java/com/zest/android/data/Category.java
package com.zest.android.data;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;
import com.zest.android.BuildConfig;

import org.greenrobot.greendao.annotation.Entity;
import org.greenrobot.greendao.annotation.Generated;
import org.greenrobot.greendao.annotation.Id;
import org.greenrobot.greendao.annotation.Keep;
import org.greenrobot.greendao.annotation.Property;
import org.parceler.Parcel;


@Parcel
public class Category {

    @Id(autoincrement = true)
    private Long id;

    @SerializedName("strCategory")
    @Expose
    private String title;

    @SerializedName("idCategory")
    @Expose
    private String categoryId;

    @SerializedName("strCategoryThumb")
    @Expose
    private String image;

    @SerializedName("strCategoryDescription")
    @Expose
    private String description;

    public Category() {
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getCategoryId() {
        return categoryId;
    }

    public void setCategoryId(String categoryId) {
        this.categoryId = categoryId;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    @Override
    public String toString() {
        return "Category{" +
                "id=" + id +
                ", title='" + title + '\'' +
                ", categoryId='" + categoryId + '\'' +
                ", image='" + image + '\'' +
                ", description='" + description + '\'' +
                '}';
    }
}

File: app/src/main/java/com/zest/android/data/Category.kt
package com.zest.android.data

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import org.greenrobot.greendao.annotation.Id


data class Category(@Id(autoincrement = true) var id: Long? = null,
                    @SerializedName("strCategory") @Expose var title: String,
                    @SerializedName("idCategory") @Expose var categoryId: String? = null,
                    @SerializedName("strCategoryThumb") @Expose var image: String? = null,
                    @SerializedName("strCategoryDescription") @Expose var description: String? = null) : Parcelable {


    constructor(parcel: Parcel) : this(
            parcel.readValue(Long::class.java.classLoader) as? Long,
            parcel.readString(),
            parcel.readString(),
            parcel.readString(),
            parcel.readString())

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeValue(id)
        parcel.writeString(title)
        parcel.writeString(categoryId)
        parcel.writeString(image)
        parcel.writeString(description)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Category> {
        override fun createFromParcel(parcel: Parcel): Category {
            return Category(parcel)
        }

        override fun newArray(size: Int): Array<Category?> {
            return arrayOfNulls(size)
        }
    }

}